djangocms_admin_style/static/js/base-admin.js

```diff
@@ -42,6 +42,23 @@
             }
         }
     });
+
+    // Back to link
+    var button = document.getElementById('header-btn');
+
+    if (window.name !== '') {
+        var object = JSON.parse(window.name);
+    }
+
+    button.addEventListener('click', function () {
```

Comment:
why click handler? easier to rewrite the button href attribute and text
that way you will keep the default link behaviour with shift + click etc